pyuvdata/uvdata/ms.py

```diff
@@ -290,105 +1786,81 @@ def read_ms(
+                    np.mean(full_antenna_positions, axis=0)
+                )
+            else:
                 warnings.warn(
                     "Telescope frame is not ITRF and telescope is not "
                     "in known_telescopes, so telescope_location is not set."
                 )
+        tb_obs.close()
 
         # antenna names
-        ant_names = tb_ant.getcol("STATION")
-        ant_diams = tb_ant.getcol("DISH_DIAMETER")
-
-        self.antenna_diameters = ant_diams[ant_diams > 0]
-
-        self.Nants_telescope = len(ant_flags[np.invert(ant_flags)])
-        test_name = ant_names[0]
-        names_same = True
-        for antnum in range(len(ant_names)):
-            if not (ant_names[antnum] == test_name):
-                names_same = False
-        if not (names_same):
-            # cotter measurement sets store antenna names in the NAMES column.
+        ant_names = np.asarray(tb_ant.getcol("NAME"))[ant_good_position].tolist()
+        station_names = np.asarray(tb_ant.getcol("STATION"))[ant_good_position].tolist()
+        self.antenna_diameters = tb_ant.getcol("DISH_DIAMETER")[ant_good_position]
+
+        # importuvfits measurement sets store antenna names in the STATION column.
+        # cotter measurement sets store antenna names in the NAME column, which is
+        # inline with the MS definition doc. In that case all the station names are
+        # the same.
+        # TODO: Potentially flip which is the default -- use ant_names if filled and
+        # unique. This does seem to fail though on "day2_TDEM0003_10s_norx_1src_1spw".
+        # See the test "test_read_ms_read_uvfits".
+        if len(station_names) != len(np.unique(station_names)):
             self.antenna_names = ant_names
         else:
-            # importuvfits measurement sets store antenna names in the STATION column.
-            self.antenna_names = tb_ant.getcol("NAME")
-        self.antenna_numbers = np.arange(len(self.antenna_names)).astype(int)
-        ant_names = []
-        for ant_num in range(len(self.antenna_names)):
-            if not (ant_flags[ant_num]):
-                ant_names.append(self.antenna_names[ant_num])
-        self.antenna_names = ant_names
-        self.antenna_numbers = self.antenna_numbers[np.invert(ant_flags)]
+            self.antenna_names = station_names
+
+        self.Nants_telescope = len(self.antenna_names)
 
         relative_positions = np.zeros_like(full_antenna_positions)
         relative_positions = full_antenna_positions - self.telescope_location.reshape(
             1, 3
         )
-        self.antenna_positions = relative_positions[np.invert(ant_flags), :]
+        self.antenna_positions = relative_positions
 
         tb_ant.close()
+
+        self._set_phased()
+
+        # set LST array from times and itrf
+        proc = self.set_lsts_from_time_array(background=background_lsts)
+
+        # CASA weights column keeps track of number of data points averaged.
```

Comment:
what code does this comment go with?

Reply:
Heh, something that got deleted during code refactoring...

Reply:
Removed in 3068e067.